apps/mobile/src/features/TxHistory/TxHistory.container.tsx

```diff
@@ -31,8 +33,29 @@ export function TxHistoryContainer() {
     }
   }
 
-  // Combine loading states
-  const isLoadingState = isFetching || isLoading || isUninitialized
-
-  return <TxHistoryList transactions={transactions} onEndReached={onEndReached} isLoading={isLoadingState} />
+  // Handle pull-to-refresh - reset the data and fetch from the beginning
+  const onRefresh = React.useCallback(async () => {
+    setIsRefreshing(true)
+    try {
+      // Refetch will reset the data and start fresh with page 1
+      await refetch()
+    } catch (error) {
+      Logger.error('Error refreshing transaction history:', error)
+    } finally {
+      setIsRefreshing(false)
+    }
+  }, [refetch])
+
```

Comment:
It would be nice to move this logic to a hook, so we can combine the infinite scroll with pull-to-refresh as well 🤔 wdyt?

Reply:
Let's abstract this when we work on the pendingTx view and need to do the same there.